util/mobile_base.hpp

```diff
@@ -0,0 +1,325 @@
+#pragma once
+
+#include <queue>
+
+#include "group.hpp"
+#include "group_command.hpp"
+#include "group_feedback.hpp"
+#include "trajectory.hpp"
+
+#include "Eigen/Dense"
+
+#include "group_manager.hpp"
+
+
+namespace hebi {
+namespace experimental {
+namespace mobile {
+
+using std::queue;
+using std::shared_ptr;
+using hebi::trajectory::Trajectory;
+
+class SE2Point {
+  public:
+  SE2Point() {}
+  SE2Point(double x, double y, double theta) : x(x), y(y), theta(theta) {}
+
+  double x{std::numeric_limits<double>::quiet_NaN()};
+  double y{std::numeric_limits<double>::quiet_NaN()};
+  double theta{std::numeric_limits<double>::quiet_NaN()};
+
+  SE2Point operator+(const SE2Point& rhs) const {
+    SE2Point out;
+    out.x = this->x + rhs.x;
+    out.y = this->y + rhs.y;
+    out.theta = this->theta + rhs.theta;
+    return out;
+  }
+
+  SE2Point operator-(const SE2Point& rhs) const {
+    SE2Point out;
+    out.x = this->x - rhs.x;
+    out.y = this->y - rhs.y;
+    out.theta = this->theta - rhs.theta;
+    return out;
+  }
+
+  void operator+=(const SE2Point& rhs) {
+    this->x += rhs.x;
+    this->y += rhs.y;
+    this->theta += rhs.theta;
+  }
+
+};
+
+using Pose = SE2Point;
+using Vel = SE2Point;
+
+Pose operator*(const Vel& val, double t) {
+  SE2Point out{val.x*t, val.y*t, val.theta*t};
+  return out;
+}
+
+class Waypoint {
+  public:
+  double t{std::numeric_limits<double>::quiet_NaN()};
+  Pose pos{};
+  Vel vel{};
+  SE2Point acc{};
+};
+
+struct CartesianGoal {
+  // Moves with the given relative body velocity for a certain
+  // number of seconds, then slows to a stop over a certain number
+  // of seconds.  Can be used for continuous replanning, e.g.
+  // mobile IO control.
+  // \param v target velocity
+  // \param cruiseTime time to drive at velocity v
+  // \param slowTime "ramp down" time
+  static CartesianGoal createFromVelocity(Vel v, double cruiseTime, double slowTime) {
+    Waypoint start;
+    start.t = 0.1;
+    start.vel = v;
+
+    Waypoint slow;
+    slow.t = cruiseTime;
+    slow.vel = v;
+
+    Waypoint stop;
+    stop.t = cruiseTime + slowTime;
+    stop.vel.x = 0;
+    stop.vel.y = 0;
+    stop.vel.theta = 0;
+
+    return createFromWaypoints({start, slow, stop}, true);
+  }
+
+  // Goes to (x, y, theta) in a certain number of seconds.
+  // \param isBaseFrame is "true" if this is relative to current body frame,
+  // and "false" for using odometry coordinates.
+  static CartesianGoal createFromPosition(Pose p, double goalTime, bool isBaseFrame) {
```

Comment:
remove `isBaseFrame` until we are sure we need it? 

Also, drop `start` here, as this is generated by the base.